scripts/build-worker.sh

```diff
@@ -10,29 +20,84 @@ fi
 # https://improbableio.atlassian.net/browse/ENG-945
 function printLog() {
     if [[ ${LOG_LOCATION-} ]]; then
-        cat "${LOG_LOCATION}" 1>&2 
+        cat "${LOG_LOCATION}" 1>&2
     fi
 }
 
 trap printLog ERR
 
 source "$(dirname "$0")/pinned-tools.sh"
+source "$(dirname "$0")/profiling.sh"
 
-echo "Building for: ${WORKER_TYPE} ${BUILD_TARGET} ${SCRIPTING_TYPE}"
+echo "Building for: ${WORKER_TYPE} ${BUILD_ENVIRONMENT} ${SCRIPTING_BACKEND}"
 
 pushd "$(dirname "$0")/../"
-  if [ ${WORKER_TYPE} == "AndroidClient" ] || [ ${WORKER_TYPE} == "MobileClient" ]; then
-      scripts/prepare-unity-mobile.sh "$(pwd)/../logs/PrepareUnityMobile.log"
-  fi
+    if [ "${WORKER_TYPE}" == "AndroidClient" ] || [ "${WORKER_TYPE}" == "MobileClient" ]; then
+        if [[ ${BUILD_TARGET_FILTER-} != "ios" ]]; then
+            scripts/prepare-unity-android.sh "$(pwd)/../logs/PrepareUnityMobile.log"
+        fi
+    fi
+
+    if [[ ${WORKER_TYPE} == "iOSClient" ]]; then
+        if ! isMacOS; then
+            echo "I can't build for iOS!"
+            exit 1
+        fi
+    fi
+
+    # The asset cache ip cannot be hardcoded and so is stored in an environment variable on the build agent.
+    # This is bash shorthand syntax for if-else predicated on the existance of the environment variable
+    # where the else branch assigns an empty string.
+    #   i.e. -
+    #   if [ -z ${UNITY_ASSET_CACHE_IP} ]; then
+    #       ASSET_CACHE_ARG="-CacheServerIPAddress ${UNITY_ASSET_CACHE_IP}"
+    #   else
+    #       ASSET_CACHE_ARG=""
+    #   fi
+    ASSET_CACHE_ARG=${UNITY_ASSET_CACHE_IP:+-CacheServerIPAddress "${UNITY_ASSET_CACHE_IP}"}
+
+    RUN_UNITY_PATH="$(pwd)/tools/RunUnity/RunUnity.csproj"
+
+    pushd "$(pwd)/../workers/unity"
+        rm -rf ./build/worker/
```

Comment:
Hmmm I wonder why we did this & is it required in BK land

Reply:
https://github.com/spatialos/gdk-for-unity-shared-ci/pull/13/commits/e099c93243af35974200acc44f387fdfde556cec

think I added it as a workaround for some rogue team city bug

probably not needed in BK